sql/testdata/datetime

```diff
@@ -42,6 +42,44 @@ statement error duplicate key value \(a\)=\(2015-08-30 03:34:45\.34567\+00:00\)
 INSERT INTO t VALUES
   ('2015-08-29 20:34:45.34567-07:00'::timestamp, '2015-08-31'::date, '35h2s'::interval)
 
+statement ok
+SET TIME ZONE 'EUROPE/ROME'
+
+statement error duplicate key value \(a\)=\(2015-08-30 03:34:45\.34567\+00:00\) violates unique constraint "primary"
+INSERT INTO t VALUES
+  ('2015-08-30 04:34:45.34567 CET'::timestamp, '2015-08-31'::date, '35h2s'::interval)
+
+statement ok
+SET TIME ZONE 'America/New_York'
+
+statement error duplicate key value \(a\)=\(2015-08-30 03:34:45\.34567\+00:00\) violates unique constraint "primary"
+INSERT INTO t VALUES
+  ('2015-08-29 22:34:45.34567 EST'::timestamp, '2015-08-31'::date, '35h2s'::interval)
+
+statement error cannot find foobar.*
+SET TIME ZONE 'foobar'
+
+statement ok
+SET TIME ZONE INTERVAL '-7h'
```

Comment:
The most common syntax I've seen in practice for time zones is '-7:00' (in mysql: `set time zone '-7:00'`; in postgres it is the cumbersome `set time zone interval '-7:00' hour to minute`). Do/could we support that?